packages/block-editor/src/components/list-view/branch.js

```diff
@@ -23,39 +23,59 @@ import { isClientIdSelected } from './utils';
  * When a block is collapsed, we do not count their children as part of that total. In the current drag
  * implementation dragged blocks and their children are not counted.
  *
- * @param {Object} block            block tree
- * @param {Object} expandedState    state that notes which branches are collapsed
- * @param {Array}  draggedClientIds a list of dragged client ids
+ * @param {Object}  block               block tree
+ * @param {Object}  expandedState       state that notes which branches are collapsed
+ * @param {Array}   draggedClientIds    a list of dragged client ids
+ * @param {boolean} isExpandedByDefault flag to determine the default fallback expanded state.
  * @return {number} block count
  */
-function countBlocks( block, expandedState, draggedClientIds ) {
+function countBlocks(
+	block,
+	expandedState,
+	draggedClientIds,
+	isExpandedByDefault
+) {
```

Comment:
It's a weird thing that the function is named countBlocks and we handle expanded state in it or depend on it ...

Reply:
Possibly. I was here prior to this PR but I take your point. 

The reason it exists is because the function tries to determine the number of currently "shown" blocks. Collapsed block nodes don't count.

Reply:
Circling back here - this is covered by this comment

https://github.com/WordPress/gutenberg/blob/799a3e93c221542c1376f9ec114c8762457db8ef/packages/block-editor/src/components/list-view/branch.js#L22-L23